netty/src/main/java/io/grpc/transport/netty/NettyServerHandler.java

```diff
@@ -287,6 +289,12 @@ private void sendResponseHeaders(ChannelHandlerContext ctx, SendResponseHeadersC
     encoder().writeHeaders(ctx, cmd.streamId(), cmd.headers(), 0, cmd.endOfStream(), promise);
   }
 
+  private void cancelStream(ChannelHandlerContext ctx, CancelServerStreamCommand cmd,
+      ChannelPromise promise) {
+    serverStream(requireHttp2Stream(cmd.streamId())).abortStream(cmd.reason(), false);
```

Comment:
right. The only reason to prefer to notify the client is to propagate the deadline exceeded status. I'll file an issue about the failing call.